examples/llm/src/models/layers/attention.py

```diff
@@ -82,11 +82,17 @@ def generate_attn_mask(
         batch_size,
         heads,
         seq_len,
+        prefix_mask=None,
         key_padding_mask=None,
         alibi=False,
         dtype=None,
     ):
 
+        if prefix_mask is not None:
+            raise NotImplementedError(
+                'This attention implementation does not support use of `prefix_lm`.'
+            )
+
```

Comment:
Is there anything that stops us from supporting prefix_mask for torch impl (non-causal)?

Reply:
My laziness. I suppose I should add support for that.

Reply:
@vchiley Actually, is it worth messing around with the torch implementation? It has a deprecation warning and is already missing support for a number of features we seem to be moving towards.

Reply:
Given the attn rewrite, all attn impls are merged into one cls.
The only version that cannot support bias is flash and an error gets raised if you pass an attn_bias to it.